sql/src/test/java/io/crate/expression/reference/sys/cluster/ClusterLicenseExpressionTest.java

```diff
@@ -0,0 +1,79 @@
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.reference.sys.cluster;
+
+import io.crate.license.DecryptedLicenseData;
+import io.crate.license.LicenseService;
+import io.crate.test.integration.CrateUnitTest;
+import org.junit.Test;
+
+import java.util.Map;
+
+import static io.crate.license.DecryptedLicenseData.EXPIRATION_DATE_IN_MS;
+import static io.crate.license.DecryptedLicenseData.ISSUED_TO;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsNull.nullValue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ClusterLicenseExpressionTest extends CrateUnitTest {
+
+    @Test
+    public void tesExpressionValueReturnsNullForMissingLicense() {
+        LicenseService licenseService = mock(LicenseService.class);
+        when(licenseService.currentLicense()).thenReturn(null);
+
+        ClusterLicenseExpression licenseExpression = new ClusterLicenseExpression(licenseService);
+        assertThat(licenseExpression.value(), is(nullValue()));
+    }
+
+    @Test
+    public void testExpressionValueReturnsLicenseFields() {
+        LicenseService licenseService = mock(LicenseService.class);
+        when(licenseService.currentLicense()).thenReturn(new DecryptedLicenseData(3L, "test"));
+
+        ClusterLicenseExpression licenseExpression = new ClusterLicenseExpression(licenseService);
+        Map<String, Object> expressionValues = licenseExpression.value();
+        assertThat(expressionValues.size(), is(2));
+        assertThat(expressionValues.get(EXPIRATION_DATE_IN_MS), is(3L));
+        assertThat(expressionValues.get(ISSUED_TO), is("test"));
+    }
+
+    @Test
+    public void testGetChildAfterValueCallReturnsUpdatedLicenseFields() {
+        LicenseService licenseService = mock(LicenseService.class);
+        when(licenseService.currentLicense())
+            .thenReturn(new DecryptedLicenseData(3L, "test"))
+            .thenReturn(new DecryptedLicenseData(4L, "test2"))
+            .thenReturn(new DecryptedLicenseData(4L, "test2"));
+
+        ClusterLicenseExpression licenseExpression = new ClusterLicenseExpression(licenseService);
+        Map<String, Object> expressionValues = licenseExpression.value();
```

Comment:
could you please add another test where you test the getChild() without invoking the value() first? 
This was the root cause of the issue - I think this test would pass even without this fix

Reply:
no it wouldn't as the `childImplementations` map would contain what the `value` method filled in before

Reply:
your are right on this! 
What I meant is that the test testGetChild() that you now added would fail without the fix - bad wording, sorry :)